lib/elixir/lib/enum.ex

```diff
@@ -1453,51 +1453,74 @@ defmodule Enum do
 
-  If multiple elements are considered maximal, the first one that was found
-  is returned.
+  By default, the comparison is done with the `>=` sorter function.
+  If multiple elements are considered maximal, the first one that
+  was found is returned. If you want the last element considered
+  maximal to be returned, the sorter function should not return true
+  for equal elements.
 
-  Calls the provided `empty_fallback` function and returns its value if
-  `enumerable` is empty. The default `empty_fallback` raises `Enum.EmptyError`.
+  If the enumerable is empty, the provided `empty_fallback` is called.
+  The default `empty_fallback` raises `Enum.EmptyError`.
 
   ## Examples
 
       iex> Enum.max([1, 2, 3])
       3
 
-      iex> Enum.max([], fn -> 0 end)
-      0
-
   The fact this function uses Erlang's term ordering means that the comparison
   is structural and not semantic. For example:
 
       iex> Enum.max([~D[2017-03-31], ~D[2017-04-01]])
       ~D[2017-03-31]
 
-  In the example above, `max/1` returned March 31st instead of April 1st
-  because the structural comparison compares the day before the year. This
-  can be addressed by using `max_by/3` and by relying on structures where
-  the most significant digits come first. In this particular case, we can
-  use `Date.to_erl/1` to get a tuple representation with year, month and day
-  fields:
+  In the example above, `max/2` returned April 1st instead of March 31st
+  because the structural comparison compares the day before the year.
+  For this reason, most structs provide a "compare" function, such as
+  `Date.compare/2`, which receives two structs and returns `:lt` (less than),
+  `:eq` (equal), and `:gt` (greather than). If you pass a module as the
+  sorting function, Elixir will automatically use the `compare/2` function
+  of said module:
 
-      iex> Enum.max_by([~D[2017-03-31], ~D[2017-04-01]], &Date.to_erl/1)
+      iex> Enum.max([~D[2017-03-31], ~D[2017-04-01]], Date)
       ~D[2017-04-01]
 
-  For selecting a maximum value out of two consider using `Kernel.max/2`.
+  Finally, if you don't want to raise on empty enumerables, you can pass
+  the empty callback:
+
+      iex> Enum.max([], &>=/2, fn -> 0 end)
+      0
 
   """
-  @spec max(t, (() -> empty_result)) :: element | empty_result when empty_result: any
-  def max(enumerable, empty_fallback \\ fn -> raise Enum.EmptyError end)
-      when is_function(empty_fallback, 0) do
-    aggregate(enumerable, &Kernel.max/2, empty_fallback)
+  @spec max(t, (element, element -> boolean) | module(), (() -> empty_result)) ::
+          element | empty_result
+        when empty_result: any
+  def max(enumerable, sorter \\ &>=/2, empty_fallback \\ fn -> raise Enum.EmptyError end) do
+    {sorter, empty_fallback} = max_sort_fun(sorter, empty_fallback)
+    aggregate(enumerable, sorter, empty_fallback)
   end
 
+  # TODO: Deprecate me on 1.14
+  defp max_sort_fun(empty_fallback, default_empty_callback)
```

Comment:
There are other occurrences in the lines below.

```suggestion
  defp max_sort_fun(empty_fallback, default_empty_fallback)
```